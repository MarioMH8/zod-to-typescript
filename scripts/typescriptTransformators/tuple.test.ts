import { ZodTypescriptTransformator } from "@scripts/index";
import { z as zod } from "zod";

describe("tuple", () => {
	it("no rest", () => {
		const result = ZodTypescriptTransformator.convert(zod.tuple([zod.string(), zod.number()]));

		expect(result).toMatchSnapshot();
	});

	it("with rest", () => {
		const result = ZodTypescriptTransformator.convert(zod.tuple([zod.string(), zod.number()]).rest(zod.null()));

		expect(result).toMatchSnapshot();
	});
});
