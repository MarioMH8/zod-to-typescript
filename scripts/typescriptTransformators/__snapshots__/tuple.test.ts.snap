// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tuple > no rest 1`] = `
"type Zod2ts_0_duplojs = [
    string,
    number
];"
`;

exports[`tuple > with rest 1`] = `
"type Zod2ts_1_duplojs = [
    string,
    number,
    ...null[]
];"
`;
